### DISCUSSION TITLE
thread for one sample

### DISCUSSION THREAD

**Discussion by Liripo** [GreenleafLab/ArchR#1339] [Questions / Documentation]
Is there no performance loss for a single sample when I use addArchRThreads(thread = 1)？

**Comment by rcorces** [comment]
certain functions parallelize on samples. in those cases, you wont notice. But most functions parallelize on matrix chunks and you will see significant performance boosts by allowing more threads.

**Reply by Liripo** [reply]
Thank you for your reply！
But when I set up multiple processes, for example, I get an error when I use the `plotFragmentSizes` function, like issues [https://github.com/GreenleafLab/ArchR/issues/248](https://github.com/GreenleafLab/ArchR/issues/248),but I use `createArrowFiles` and I don't get this error. I don't know how to fix it, it seems to be a problem with h5 and mclapply?
